### ISSUE TITLE
After class deletion:com.orientechnologies.orient.core.index.OIndexException: Error on rebuilding the index for clusters

### ISSUE THREAD

**Issue by StarpTech** [orientechnologies/orientdb#5111]
1) I want to delete the `ShopUser` class and this error is thrown

```
com.orientechnologies.orient.core.index.OIndexException: Error on rebuilding the index for clusters: [node, de_fitness, root, fitness, shopuser, de_root, de_business]
 --> java.lang.IllegalArgumentException: Cluster 'shopuser' was not found
```

2) I try it again and this error is thrown

```
java.lang.IllegalArgumentException: Cluster 15 is null
```

but the Class is still listed.

3) Now I want to use my db like before
```
select from shopuser
```

Error:

```
java.lang.IllegalArgumentException: Cluster 15 is null
```
My db is broken. I don't use a complex db schema :( @lvca sry but this release serie is unusable. This is no issue this is the core the basic of the basics. :/ I deeply hope you get it stable. Thanks.

![schema](https://cloud.githubusercontent.com/assets/1764424/10467232/ec8db5ec-71f8-11e5-9280-d673e5bde84d.PNG)

**Comment by MarcelPitch** [comment]
Hi @lvca,

I've the same problem than @StartTech.

Could you provide us a temporary workaround in order to repair database ?

Thx and have a nice day

**Comment by luigidellaquila** [comment]
Hi guys, 

I think I have a fix for this, I'm running tests right now. If everything is fine I'll be able to push it on 2.1.x branch in a few minutes

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi guys, 

I just pushed a fix on develop branch.
The fix just prevents for this problem to happen, but maybe you need to drop and recreate indexes in the class hierarchy to solve the problem.
Please let me know if you need help on this

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Sorry, the fix is on branch 2.1.x, I'm porting it to develop right now

**Comment by luigidellaquila** [comment]
Hi guys, this is fixed in both develop and 2.1.x, please let me know if it completely fixes your use case

Thanks

Luigi

**Comment by larryaasen** [comment]
@luigidellaquila Hi Luigi, I just ran into this problem. You said "maybe you need to drop and recreate indexes in the class hierarchy". Is there an easy way to do that with a database in production? Do all indexes have to be dropped/recreated?
